sympy/concrete/expr_with_limits.py

```diff
@@ -90,7 +91,13 @@ def _process_limits(*symbols):
                 limits.append(Tuple(V))
             continue
         elif is_sequence(V, Tuple):
-            V = sympify(flatten(V))
+            new_V = V
+            if len(V) > 1:
+                if isinstance(V[1], Range):
+                    if V[1].step != 1:
+                        raise ValueError("The step of Range must be 1.")
+                    new_V = (V[0], V[1].inf, V[1].sup)
+            V = sympify(flatten(new_V))
             if isinstance(V[0], (Symbol, Idx)) or getattr(V[0], '_diff_wrt', False):
                 newsymbol = V[0]
                 if len(V) == 2 and isinstance(V[1], Interval):
```

Comment:
add here,
```python
if len(V) == 4:
    V[2:] = 0, (V[2] - V[1])//V[3] - 1, V[0]*V[3] + v[1]  # x, Range(1, 11, 2) -flat-> x, 1, 11, 2 -> x, 0, 4, 2*x + 1
```
Then back in the line after the call to `_process_limits` could appear:
```python
for i, lim in enumerate(limits):
    if len(lim) == 4:
        x, _, _, xnew = lim
        function = function.subs(x, xnew)
        limits[i].pop()  # no need xnew
```

Reply:
@smichr I have made the modifications according to your reviews. Thanks for the suggestions.